### PR TITLE
BUG: iloc misbehavior with pd.Series: sometimes returns pd.Categorical, fixes #14580

### LINKED ISSUE #14580: BUG: iloc misbehavior with pd.Series: sometimes returns pd.Categorical instead
#### A small, complete example of the issue

```python
# Your code here
>>> type(pd.Series([1,2,3]).astype('category').iloc[0:1])
<class 'pandas.core.series.Series'>

>>> type(pd.Series([1,2,3]).astype('category').iloc[np.array([0, 1])])
<class 'pandas.core.categorical.Categorical'>

```

#### Expected Output
Both should return a pandas Series object

#### Output of ``pd.show_versions()``

<details>
pandas version = 0.18

</details>

### PR DESCRIPTION
**Pull request by nathalier**
- [x] closes #xxxx
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

closes https://github.com/pandas-dev/pandas/issues/14580

### CODE REVIEW

File: pandas/tests/series/test_indexing.py [hunk 1]
```diff
@@ -409,6 +409,23 @@ def test_getitem_callable(self):
         result = s[lambda x: [True, False, True, True]]
         tm.assert_series_equal(result, s.iloc[[0, 2, 3]])
 
+    def test_getitem_category_type(self):
```
Comment (jreback) [inline pandas/tests/series/test_indexing.py]: put this in the test_categorical.py instead

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -1595,6 +1595,15 @@ def _get_slice_axis(self, slice_obj, axis=0):
         else:
             return self.obj.take(slice_obj, axis=axis, convert=False)
 
+    def _get_list_axis(self, key_list, axis=0):
+
```
Comment (jreback) [inline pandas/core/indexing.py]: can you add a comment on what this is doing

Comment (nathalier) [inline pandas/core/indexing.py]: Could you give a hint on phrasing please?
Would "return Series values by list or array of integers" be clear enough?

Comment (jreback) [inline pandas/core/indexing.py]: so ideally add a Paramaters/Returns section (I know we don't have docs for other of the internal functions, but have to start :))

yes  but be clear that this is a posiitional indexer

**Comment by nathalier** [comment]
It looks like appveyor CI check failed due to connectivity problem. Is it possible to restart it?

**Comment by jorisvandenbossche** [comment]
@nathalier Can you add a whatsnew note for this?

File: pandas/tests/test_categorical.py [hunk 3]
```diff
@@ -54,6 +54,24 @@ def test_getitem_listlike(self):
...
 
+    def test_getitem_category_type(self):
+        # GH 14580
+        # test iloc() on Series with Categorical data
+
+        s = pd.Series([1, 2, 3]).astype('category')
+
+        # get slice
+        result = s.iloc[0:1]
+        self.assertEqual(type(result), type(s))
```
Comment (jreback) [inline pandas/tests/test_categorical.py]: can you use ``tm.assert_series_equal`` for these (IOW, construct the expected series)

**Comment by nathalier** [comment]
@jorisvandenbossche I added whatsnew to 0.19.2. Is it OK?

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**